spring-xd-dirt/src/main/java/org/springframework/xd/dirt/rest/XDController.java

```diff
@@ -204,7 +207,16 @@ private void enhanceWithDeployments(Page<D> page, PagedResources<R> result) {
 					((DeployableResource) definitionResource).setStatus(DeploymentUnitStatus.State.undeployed.toString());
 				}
 			}
-			Assert.state(!deployedInstances.hasNext(), "Not all instances were looked at");
+			if (deployedInstances.hasNext()) {
+				final List<String> uninspectedInstanceNames = new ArrayList<String>();
+
+				while (deployedInstances.hasNext()) {
+					final BaseInstance<D> uninspectedInstance = deployedInstances.next();
+					uninspectedInstanceNames.add(uninspectedInstance.getDefinition().getName());
+				}
+				throw new IllegalStateException("Not all instances were looked at: "
```

Comment:
In general, the way the lookups are done using iterator feels odd. I chose to remove the pure Assert - in case of an error I wanted to provide some more information `Not all instances were looked at` is not that helpful.